Symfony/CS/Fixer/Contrib/PhpdocOrderFixer.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\DocBlock\DocBlock;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Graham Campbell <graham@mineuk.com>
+ */
+class PhpdocOrderFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($tokens->findGivenKind(T_DOC_COMMENT) as $token) {
+            $content = $token->getContent();
+            // move param to start, return to end, leave throws in the middle
+            $content = $this->moveParamAnnotations($content);
+            // we're parsing the content again to make sure the internal
+            // state of the dockblock is correct after the modifications
+            $content = $this->moveReturnAnnotations($content);
+            // persist the content at the end
+            $token->setContent($content);
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Annotations in phpdocs should be ordered so that param annotations come first, then throws annotations, then return annotations.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must be run before the PhpdocSeperationFixer
```

Comment:
The order fixer must be run first otherwise we're separating the annotations while they're in the wrong order, then ordering them which might break the separation.